src/dma.rs

```diff
@@ -316,7 +317,14 @@ macro_rules! dma {
                             // we need a fence here for the same reason we need one in `Transfer.wait`
                             atomic::compiler_fence(Ordering::Acquire);
 
-                            (self.buffer, self.payload)
+                            // NOTE(unsafe) There is no panic branch between getting the resources
+                            // and forgetting `self`.
+                            unsafe {
+                                let buffer = ptr::read(&self.buffer);
+                                let payload = ptr::read(&self.payload);
+                                mem::forget(self);
+                                (buffer, payload)
+                            }
```

Comment:
I'm probably missing something here, but why is this required? It seems like this would be the smae as just moving `self.buffer` and `self.payload` into the tuple directly?

Reply:
`Transfer` now implements `Drop`, so we can't move out of its fields.

Reply:
Ah right. Could you add a comment clarifying that in the code? Seems like something that will come up in the future